[shenchenzizoe] iP

shenchenzizoe opened the pull request:
## DukePro

> "An hour of planning can save you 10 hours of doing. " -Dale Carnegie([source](https://medium.com/@catherine_40404/an-hour-of-planning-can-save-you-10-hours-of-doing-part-1-d75d8140ad7e))

DukePro can be your personal task-management assistant! It is:
- simple text-based
- easy to use
- fast

All you need to do is,

1. download it from [here](https://github.com/shenchenzizoe/ip/releases/download/A-Jar/duke.jar).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE**!

Features:

- [x] Managing todo tasks
- [x] Managing deadlines
- [x] Managing events

If you Java programmer, you can use it to practice Java too. Here's the `main` method:
```
public static void main(String[] args) throws IOException {
      try {
        new Duke("data/duke.txt").run();
      } catch (FileNotFoundException e) {
        File data = new File("data");
        data.mkdirs();
        File duke = new File("data/duke.txt");
        new Duke(duke.getPath()).run();
      }
    }
```

Commit:
Add Gradle support

shenchenzizoe committed:
finish level 2

shenchenzizoe committed:
finish level 3

shenchenzizoe committed:
finish level 4

shenchenzizoe committed:
finish level 4

shenchenzizoe committed:
finish level 5

shenchenzizoe committed:
finish level 6

shenchenzizoe committed:
Automated Text UI Testing

tangyuantyy commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,21 @@
+public class Deadline extends Task1{
+  public String time;
```
```suggestion
  private String time;
```

tangyuantyy commented on src/main/java/DontKnow.java:
```diff
@@ -0,0 +1,7 @@
+public class DontKnow extends  DuckException {
```
```suggestion
public class DontKnow extends  DukeException {
```

tangyuantyy commented on src/main/java/DuckException.java:
```diff
@@ -0,0 +1 @@
+public class DuckException extends Exception{ }
```
```suggestion
public class DukeException extends Exception{ }
```

tangyuantyy replied:
Should it be named as DukeException since your main class is named Duke?

tangyuantyy commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,43 @@
+  public void mark() {
+    this.mark = true;
+  }
+  public void demark() {
```
```suggestion
  public void unmark() {
```

tangyuantyy replied:
Should it be named as unmark because demark means to demarcate and it might be misleading?

tangyuantyy commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,188 @@
+        Task[] tasks = new Task[100];
+        int index = 0;
+        while (!str.equals("bye")) {
+            if(str.equals("list")) {
```
Should there be a space between if and ( ?

tangyuantyy reviewed (commented):
Overall, I found your code a little complicated as some of the classes are quite long. I also noted that there are some coding standard violations (e.g. naming and layout). Maybe you could rename the classes and variables so that it is easier to understand, and consider making everything cleaner!

SFCoding123 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,188 @@
+import java.io.IOException;
+import java.io.InputStreamReader;
+
 public class Duke {
```
Maybe can use switch case to improve readability.

SFCoding123 commented on src/main/java/Duke2.java:
```diff
@@ -0,0 +1,244 @@
+import java.io.InputStreamReader;
+import java.util.ArrayList;
+
+public class Duke2 {
```
Maybe the naming of the class can be more distinct with Duke?

SFCoding123 commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,43 @@
+public class Task {
+  public String string;
+  public Boolean mark;
+  public Boolean todo;
+  public Boolean deadline;
+  public Boolean event;
+  public String time;
+  public String startTime;
+  public String endTime;
```
The name of the fields can be improved to follow the Name well guideline. Such as string can be replace with str and mark can be replace with isMark.

SFCoding123 commented:
Overall, I like the flow of your code. Maybe can just clean up a bit on the naming to improve the readability.

shenchenzizoe committed:
Finish level 7

shenchenzizoe committed:
Finish level 8

shenchenzizoe committed:
Finish level 8

shenchenzizoe committed:
finish A-MoreOOP

shenchenzizoe committed:
finish A-Packages

shenchenzizoe committed:
Merge remote-tracking branch 'origin/add-gradle-support'

shenchenzizoe committed:
Merge remote-tracking branch 'origin/add-gradle-support'

shenchenzizoe committed:
finished A-JUnit

shenchenzizoe committed:
gradle issue

shenchenzizoe committed:
gradle issue solved

shenchenzizoe committed:
solved delete issues

shenchenzizoe committed:
adjusted folder orders

shenchenzizoe committed:
solve the creating data file issue

shenchenzizoe committed:
finish A-JavaDoc

shenchenzizoe committed:
finish A-JavaDoc

shenchenzizoe committed:
finish A-CodingStandard

shenchenzizoe committed:
finish level 9

shenchenzizoe committed:
merged level 9

shenchenzizoe committed:
merged level 9

shenchenzizoe committed:
Merge branch 'branch-A-JavaDoc'

shenchenzizoe committed:
Merge branch 'branch-A-CodingStandard'

shenchenzizoe committed:
Merge commit 'b7b558cf16b8e1d0603a74f4c4972d728c4204ed'

shenchenzizoe committed:
level 9 find

shenchenzizoe committed:
Merge branch 'branch-Level-9'

shenchenzizoe committed:
use checkstyle

shenchenzizoe committed:
checkstyle 2

shenchenzizoe committed:
Merge branch 'branch-A-checkstyle'

# Conflicts:
#	src/main/java/Deadline.java

shenchenzizoe committed:
implement GUI

shenchenzizoe committed:
implement GUI

shenchenzizoe committed:
delete original Main class

The functions of the Main class is substituted, so delete it to make the code cleaner.

deleting Main as a whole class, while naming the new main class Main2 before the deletion

It ensures the code has basic functions before new structures are fully implemented.

uploaded the corresponding jar file after this.

shenchenzizoe committed:
finish first version of user guide

shenchenzizoe committed:
change dialogBox setting

shenchenzizoe committed:
A-assertions

shenchenzizoe committed:
A-codequality

shenchenzizoe committed:
Merge pull request #2 from shenchenzizoe/branch-A-assertions

A-assertions

shenchenzizoe committed:
Merge pull request #3 from shenchenzizoe/branch-A-codequality

A-codequality

shenchenzizoe committed:
A-extension

shenchenzizoe committed:
Merge remote-tracking branch 'origin/master'

shenchenzizoe committed:
update user guide

shenchenzizoe committed:
update ui.png

shenchenzizoe committed:
debug mark command

shenchenzizoe committed:
debug for corner case
“event Seth /from /by 2023-10-12 1000”

shenchenzizoe committed:
saved list issue for gui

shenchenzizoe committed:
add another extension

shenchenzizoe committed:
add another extension

shenchenzizoe committed:
update UG

shenchenzizoe committed:
Update README.md

shenchenzizoe committed:
Update README.md

shenchenzizoe committed:
Update README.md
